storm-core/src/clj/org/apache/storm/daemon/supervisor.clj

```diff
@@ -425,6 +433,13 @@
          ". State: " state
          ", Heartbeat: " (pr-str heartbeat))
         (shutdown-worker supervisor id)))
+
+    (doseq [storm-id all-downloaded-storm-ids]
```

Comment:
@harshach 
I agree that it would be nice to keep topology code until topology gets killed, but since this is same to what it was, I guess we can handle this as new issue for improvement.
Btw, `assignment-snapshot` seems hiding no topology info (I mean killed) vs no assignment. We should let assignment-snapshot distinguish both.